src/pages/BrowseNftsPage.tsx

```diff
@@ -11,6 +12,15 @@ interface Params {
 
 const Container = styled.div``;
 
+const NftName: React.FunctionComponent<{ tokenUri: string }> = props => {
+  // eslint-disable-next-line react/prop-types
+  const { metadata } = useMetadata(props.tokenUri);
```

Comment:
you can slam this hook in components (eg, 1 component for-each NFT in a grid view), it will handle concurrency / backpressure (eventually if needed)